pkg/core/region.go

```diff
@@ -194,6 +195,54 @@ func RegionFromHeartbeat(heartbeat *pdpb.RegionHeartbeatRequest, opts ...RegionC
 	return region
 }
 
+// RegionFromForward constructs a Region from forwarded region heartbeat.
+func RegionFromForward(heartbeat *schedulingpb.RegionHeartbeatRequest, opts ...RegionCreateOption) *RegionInfo {
```

Comment:
RegionFromForward and RegionFromHeartbeat seem to be different only `regionKvSize := heartbeat.GetApproximateKvSize() / units.MiB` here.

Reply:
request type is different